net/SessionManager.c

```diff
@@ -261,8 +261,9 @@ static Iface_DEFUN incomingFromSwitchIf(struct Message* msg, struct Iface* iface
         uint8_t* herKey = caHeader->handshake.publicKey;
         uint8_t ip6[16];
         // a packet which claims to be "from us" causes problems
-        if (!AddressCalc_addressForPublicKey(ip6, herKey)) {
-            Log_debug(sm->log, "DROP Handshake with non-fc key");
+        AddressCalc_addressForPublicKey(ip6, herKey);
+        if (!AddressCalc_validAddress(ip6)) {
```

Comment:
Maybe it should not be in this PR.
I'm trying to make cjdns addresses to be distinguishable from "plain" fc00::/8 (960df05). In this case I'm introducing an extension. For backward capability `valid!=distinguishable(cjdns-specific)`. So for generation (`addressForPublicKey`) I use `distinguishable` address, that is too strict to be `valid` (for now). That why this change is needed.